File: common/src/main/java/com/nivasafinance/common/context/UserContext.java
package com.nivasafinance.common.context;

public class UserContext {
    private static final ThreadLocal<String> usernameThreadLocal = new ThreadLocal<>();

    private UserContext() {
        // Private constructor to prevent instantiation
    }

    public static void setUsername(String username) {
        usernameThreadLocal.set(username);
    }

    public static String getUsername() {
        return usernameThreadLocal.get();
    }

    public static void clear() {
        usernameThreadLocal.remove();
    }
}

File: common/src/main/java/com/nivasafinance/common/base/context/UserContext.kt
package com.nivasafinance.common.base.context

import com.nivasafinance.common.base.model.UserInfo

object UserContext {
    private val threadLocal = ThreadLocal<UserInfo>()

    fun setUserInfo(userInfo: UserInfo) {
        threadLocal.set(userInfo)
    }

    fun getUserInfo(): UserInfo? {
        return threadLocal.get()
    }

    fun clear() {
        threadLocal.remove()
    }
}
